Automattic/newspack-plugin
# feat(corrections): add corrections and clarifications behind feature flag

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://app.asana.com/0/1208894036683495/1208945901567236 and https://app.asana.com/0/1208894036683495/1208945901567238

This PR adds the corrections feature behind the `NEWSPACK_CORRECTIONS_ENABLED` feature flag. Note this is a quick and simple implementation based on an already existing custom plugin strictly for the purpose of getting ready for a site migration. 

**Editor:**
![Screenshot 2025-01-08 at 16 35 44](https://github.com/user-attachments/assets/42878377-c358-4c71-868d-d2bfc40c5b11)

**Frontend:**
![Screenshot 2025-01-08 at 16 34 54](https://github.com/user-attachments/assets/f8c1d1bf-ec8f-45da-a520-b4d4133f0559)


### How to test the changes in this Pull Request:

1. Enable the corrections feature flag in wp-config (`NEWSPACK_CORRECTIONS_ENABLED`)
2. Edit any post as admin
3. Scroll to the bottom of the content editor and verify the corrections metabox is present
4. Select the activate corrections checkbox, select a location, and add some corrections
5. As a reader visit the post on the front end
6. Verify the corrections are added to the post
7. As admin again, select a different corrections location
8. As reader again, visit the post and verify the corrections have moved
9. As admin again, uncheck the activate corrections checkbox and save
10. As reader again, visit the post and verify the corrections are NOT added to the post
12. As admin, delete a revision and save
13. As reader, verify the revisions appear as expected on the frontend
14. As admin, remove the feature flag from wp-config and edit any post
15. Verify corrections are no longer present

### Other information:
